### DISCUSSION TITLE
Add support for other MDM's

### DISCUSSION THREAD

**Discussion by bgatestmg** [macadmins/SupportCompanion#10] [Ideas]
Support for other MDM's like SimpleMDM, JamF, Get the profile to configure SupportCompanion in Profile Creators manifests

**Comment by almenscorner** [comment]
Would love to add support for additional MDMs with actions and application install progress. To do so I would need help to understand what to look for and what actions might be of interest as I only have access to an Intune env. In the latest release I also added support for adding your own custom actions to the tray menu: https://github.com/macadmins/SupportCompanion/releases/tag/v1.0.1.80720

As for Profile Creator I will look into creating a manifest!

**Reply by almenscorner** [reply]
Submitted PR for adding Support Companion manifest: https://github.com/ProfileCreator/ProfileManifests/pull/676

**Reply by bgatestmg** [reply]
Rad! thank you good sir, i would be happy to outline some actions that would be awesome to see.

**Reply by almenscorner** [reply]
Creating profiles using iMazing or ProfileCreator is now possible. For the actions, would it be enough to have the custom actions in the tray menu and then focus on the app install progress? I think that is the main part any way :) 
![Screenshot 2024-05-30 at 16 49 35](https://github.com/macadmins/SupportCompanion/assets/78877636/a70bcc5e-c02d-404d-a3e1-dabd8cdc0188)

**Reply by brngates98** [reply]
i think thats a good way to go yea

**Reply by brngates98** [reply]
![Screenshot 2024-05-30 at 11 08 30 AM](https://github.com/macadmins/SupportCompanion/assets/59444489/2c6ce52e-0411-49de-94e0-4929e0efbb78)
just got the update for the manifests will play with it and see how it works :)

**Reply by almenscorner** [reply]
So for SimpleMDM or Jamf for example, how can I retrieve the list of pending app updates/installed apps etc.?

**Reply by brngates98** [reply]
So SimpleMDM has an API that could be queried for this information but i dont think that would be the best way, maybe though?
https://api.simplemdm.com/#list-for-device

**Reply by almenscorner** [reply]
What I want is to gather this info locally on the device so there is no need to authenticate to an API and reach out externally to get the info

**Reply by brngates98** [reply]
So for that piece SimpleMDM utilizes a hosted munki solution to be able to deploy apps, that are not in VPP. however i am deploying most of my apps initially with Baseline and then allowing munki to manage the updates to them. at least thats my workflow.
However i have been having a bug with getting munki to manage the updates to these(still learning)

However we have a small collection of apps we allow to be installed self service through munki and the ones i did install are visible

![Screenshot 2024-05-30 at 12 35 02 PM](https://github.com/macadmins/SupportCompanion/assets/59444489/0c19f639-6355-475a-a01a-1b367fe3c9c1)

**Reply by almenscorner** [reply]
So question then is if they store the states of the apps it manages updates for somewhere like Munki does

**Reply by brngates98** [reply]
> So question then is if they store the states of the apps it manages updates for somewhere like Munki does

let me ask the simplemdm team i bug them with things like this alot

**Reply by brngates98** [reply]
fired a question off: https://macadmins.slack.com/archives/C4HJ6U742/p1717092021368279

**Comment by Mj2711** [comment]
Is there any way to block the control center in jamf already tried and failed on the following available solutions--
1. ran a bash command for disabling it --- failed
2. tried block through config profile and there is no control center blockage payload
3. tried scripter ---failed



anyone know solution to it -----
![Image](https://github.com/user-attachments/assets/624728b5-46fc-4452-ae1b-abd52c644235)

**Reply by Mj2711** [reply]
there is direct paylaod available for blocking the control centre in mosyle but in jamf its pain in arm.

**Comment by TechTrekkie** [comment]
It would be great to have this available for Jamf. For the application patching, it would probably be possible to use the Jamf API to pull from patch management software titles which should be able to provide the patch compliance for each app. 
There's also jamf app installers but I'm not sure if there's anything in the API to populate the status of those, I'd have to check.

**Reply by almenscorner** [reply]
I'd be happy to add support, just need help as I don't have access to a console or the API myself :)

**Comment by almenscorner** [comment]
FYI, support for Jamf has been added in the latest pre-release
